script.js

```diff
@@ -1,4 +1,238 @@
+      gameMode = 'Player 1 Variable Dice Mode';
+    }
+    else { myOutputValue = 'Please enter "normal" or "variable" to select mode.';
+    }
+    return myOutputValue;
+  }
+
+  // Ask player1 how many dice they want to play with
+  if (gameMode == 'Player 1 Variable Dice Mode') {
+    if (isNaN(Number(input)) == false && input != '') {
+      numOfDice = input;
+      // List the index position of the dice values in order using the helper function
+      // Match it to Player 1
+      dicePosition = getDicePosition(numOfDice);
+      p1DicePosition = dicePosition;
+      // List the dice values using the helper function
+      // Match it to Player 1
+      diceValues = getNumOfDiceAndRoll(numOfDice);
+      p1DiceValues = diceValues;
+      gameMode = 'Player 1 Enter Positions';
+      // Output Player 1 dice values and prompt index question.
+      myOutputValue = 'Player 1, you have decided to <br>play with ' + numOfDice + ' dice. <br><br>You rolled: <br>---- dice: ' + p1DiceValues + '<br>position: ' + p1DicePosition + '<br><br>Enter the positions of the numbers you want.';
+    } else {
+      myOutputValue = invalidMessage;
+    }
+    return myOutputValue;
+  }
+
+  if (gameMode == 'Player 1 Enter Positions') {
+  // Check if input is a number and is not an empty string.
+    if (isNaN(Number(input)) == false && input != '') {
+      // Put input into helper function to generate player's score
+      generatePlayerScore(input);
+      player1Score = Number(playerScore);
+      // Add score to accumulate total score
+      totalp1Score += player1Score;
+      myOutputValue = 'Player 1, your score is: ' + player1Score + ". Player 2's turn. Click submit to roll.";
+      gameMode = 'Player 2 Variable Dice Mode';
+    }
+    else {
+      myOutputValue = invalidMessage;
+    }
+    // Empty the global variable of helper function output to create a new value for player 2.
+    diceValues = [];
+    dicePosition = [];
+    return myOutputValue;
+  }
+
+  if (gameMode == 'Player 2 Variable Dice Mode') {
+    // List the dice index using p1DiceValues with the helper function
+    // Match it to Player 2
+    getDicePosition(p1DiceValues.length);
+    p2DicePosition = dicePosition;
+    // List the dice values using p1DiceValues with the helper function
+    // Match it to Player 2
+    getNumOfDiceAndRoll(p1DiceValues.length);
+    p2DiceValues = diceValues;
+    gameMode = 'Player 2 Enter Position';
+    // Output Player 2 dice values and prompt index question.
+    myOutputValue = 'Player 2, you have decided to <br>play with ' + numOfDice + ' dice. <br><br>You rolled: <br>---- dice: ' + p2DiceValues + '<br>position: ' + p2DicePosition + '<br><br>Enter the positions of the numbers you want.';
+    return myOutputValue;
+  }
+
+  if (gameMode == 'Player 2 Enter Position') {
+    // Check if input is a number and is not an empty string.
+    if (isNaN(Number(input)) == false && input != '' && input.length == p2DiceValues.length) {
```

Comment:
Great validation!

Reply:
Instead of `isNaN(Number(input)) == false`, you can use `!isNaN(Number(input))`, which is more succinct and conventional